packages/models/src/crypto-assets/crypto-asset-info.model.ts

```diff
@@ -1,68 +1,114 @@
 import { InscriptionMimeType } from './bitcoin/inscription.model';
 
+export const CryptoAssetCategories = {
+  fungible: 'fungible',
+  collectible: 'collectible',
+} as const;
+export const CryptoAssetTypes = {
+  nativeBtc: 'nativeBtc',
+  nativeStx: 'nativeStx',
+  sip10: 'sip10',
+  brc20: 'brc20',
+  src20: 'src20',
+  stx20: 'stx20',
+  rune: 'rune',
+  stamp: 'stamp',
+  sip9: 'sip9',
+  inscription: 'inscription',
+} as const;
+export type CryptoAssetCategory = keyof typeof CryptoAssetCategories;
+export type CryptoAssetType = keyof typeof CryptoAssetTypes;
+
 export interface BaseCryptoAssetInfo {
+  readonly assetCategory: CryptoAssetCategory;
+  readonly assetType: CryptoAssetType;
+  readonly name: string;
+}
+
+// Fungible asset types
+interface BaseFungibleCryptoAssetInfo extends BaseCryptoAssetInfo {
+  readonly assetCategory: 'fungible';
+  readonly symbol: string;
   readonly decimals: number;
   readonly hasMemo: boolean;
 }
-
-export interface BtcCryptoAssetInfo extends BaseCryptoAssetInfo {
+export interface BtcCryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'nativeBtc';
   readonly name: 'bitcoin';
   readonly symbol: 'BTC';
 }
-
-export interface StxCryptoAssetInfo extends BaseCryptoAssetInfo {
+export interface StxCryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'nativeStx';
```

Comment:
This is a bit of a side-enhancement - But I wanted to build out the `CryptoAssetInfo` model structure a little bit, which does help within the `MarketDataService`... In general, by adding additional context to types via `assetCategory` and `assetType` fields, we can build out a more robust inheritance structure and leverage discriminated unions for collective types.

Using those types we can then distinguish between different assets and categories (e.g. collectibles vs fungible) and make assumptions in both UI Components and Services (e.g. rendering CollectibleAsset vs FungibleAsset). 

The [`MarketDataService` shows how](https://github.com/leather-io/mono/pull/565/files#diff-f9fa9619f38b7054111a03f1505088fbdb37d7d30ec848bbe8f1aa8f55fba70d) we can leverage these discriminated types in interesting ways, including the top-level `getMarketData` call, which accepts a more generic `FungibleCryptoAssetInfo` asset and the subsequent switch statement, which handles different fungible asset types separately.

@fbwoolf Interested to hear your thoughts!

Reply:
I def wouldn't introduce the category `collectible` here? We have typ always used `fungible` and `non-fungible` if it is nec to add it. Also, I don't think I feel it is nec to use `native...` instead of just `btc` and `stx`? We know those are the chain cryptocurrencies.

Reply:
`Collectible` is just a nice UI word imo.

Reply:
Tbh, a lot of these changes we just refactored out, so curious to understand why they are nec again? 🤔 Maybe it is helpful for your work, so not opposed if helpful.

Reply:
@fbwoolf Thanks for the feedback.

IMO "collectible" and "non-fungible" are pretty interchangeable. I went with "collectible" here just for consistency with other parts of the codebase.

This is probably a bigger discussion than market data, but the goal of restructuring the `CryptoAssetInfo` model is to build out a more descriptive type hierarchy and more accurately reflect the actual domain. 

It's not exactly necessary, however it does help in the market data service (as well as potentially many other services & UI components) by allowing us to leverage discriminated unions and organize / manage assets based on shared characteristics at different levels of the hierarchy (and not just rely on individual asset types + separate service calls for each).

For example, [`MarketDataService.getMarketData()`](https://github.com/leather-io/mono/pull/565/files#diff-f9fa9619f38b7054111a03f1505088fbdb37d7d30ec848bbe8f1aa8f55fba70d) can take any `FungibleCryptoAssetInfo` and handle it appropriately, using discriminators for BTC, Runes, and other fungible assets.

I could also see this setup enabling higher-level asset handling in UI components for categories like CollectibleAsset, which could render various types (e.g., stamp, sip9, inscription) without hard-coded handling for each individual asset.

Or the account balance work (cc @pete-watters  @kyranjamie) having a full list of account assets in a single array (or `fungibles[]` and `collectibles[]`) that can be handled collectively or further discriminated using the type field.

It also seems like we have a lot of one-off types throughout the codebase, which makes it more difficult to maintain consistency. By improving our models we could make them more useful and extensible - reducing the need for components and services to adapt types independently.

Reply:
I think `collectible` is a nice word and good for codebase consistency and keeping a similar language in services and UI. I don't think it's a big issue to use `collectible` or `non-fungible` but I like this classification of things.

Overall, I think this kind of type level categorisation is a good idea as it will help eliminate some of the hardcoded logic we need to add for handling individual assets as you mention. 

`assetType: 'nativeStx'` will help in the UI for example where we will need to show the asset chain and another sub avatar in non assets that aren't BTC / STX.

![Screenshot 2024-10-31 at 10 11 05](https://github.com/user-attachments/assets/62ac828c-4839-4387-b260-def6b82550e9)

Initially, I would have liked a list of assets in a single array `fungibles[]` / `collectibles[]` as my idea was to fetch this and then feed it into dumb widgets which then map over items in dumb UI components. 

It seems an alternative approach we want to take is to instead have more granular smart components responsible for fetching their own data. 

For example, I was getting the data then making a list of `tokens` with a serialiser:
```
const tokens : [{
 ticker: 'BTC', 
 balance: 0
},{
 ticker: 'STX', 
 balance: 0
},
]

.... 

<TokensWidget tokens={tokens} />
```

The pattern we are going with instead is something like:
```
function BtcBalances() {
  const {balance, fiatBalance} = useGetBtcBalance(); 
 return (<Balance balance={balance} fiatBalance={fiatBalance} />)
}

....
   <TokensWidget> 
      <BtcBalances />
      <StxBalances />
   </TokensWidget>
```

Reply:
I'd like to understand how the new types assist discriminating unions. These types already use string literals for name, such that they can discriminate.

Where does `assetType` differ from `name`? The name effectively describes the type here. So why do we need both?

Reply:
I think he is using both due to this https://github.com/leather-io/mono/pull/565/files#r1823194310